File: src/main/java/de/tectoast/emolga/utils/sql/base/columns/IntColumn.java
package de.tectoast.emolga.utils.sql.base.columns;

import de.tectoast.emolga.utils.sql.base.DataManager;

import java.sql.ResultSet;
import java.sql.SQLException;

public class IntColumn extends SQLColumn<Integer> {

    public IntColumn(String name, DataManager manager) {
        super(name, manager);
    }

    @Override
    public Integer getValue(ResultSet set) {
        try {
            return set.getInt(name);
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;
    }

    @Override
    public void update(ResultSet set, Object value) {
        try {
            set.updateInt(name, (Integer) value);
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }

    @Override
    public Integer retrieveValue(SQLColumn<?> checkcolumn, Object checkvalue) {
        return manager.read(manager.select(checkcolumn.check(checkvalue), this), rs -> {
            if (rs.next()) {
                return getValue(rs);
            }
            return -1;
        });
    }

    public void increment(SQLColumn<?> checkcolumn, Object checkvalue) {
        Integer val = retrieveValue(checkcolumn, checkvalue);
        if(val == null) val = 1;
        else val += 1;
        this.manager.editOneValue(checkcolumn, checkvalue, this, val);
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/sql/base/columns/IntColumn.kt
package de.tectoast.emolga.utils.sql.base.columns

import de.tectoast.emolga.utils.sql.base.DataManager
import java.sql.ResultSet

class IntColumn(name: String, manager: DataManager) : SQLColumn<Int?>(name, manager) {
    override fun getValue(set: ResultSet): Int {
        return set.getInt(name)
    }

    override fun retrieveValue(checkcolumn: SQLColumn<*>, checkvalue: Any): Int? {
        return DataManager.read<Int?>(manager.select(checkcolumn.check(checkvalue), this)) { rs: ResultSet ->
            if (rs.next())
                getValue(rs)
            else
                null
        }
    }

    fun increment(checkcolumn: SQLColumn<*>, checkvalue: Any) {
        manager.addStatisticsSpecified(checkvalue.toString(), listOf(checkcolumn, this), 1)
    }
}